docker-stacks/all-spark-notebook/Dockerfile

```diff
@@ -0,0 +1,16 @@
+# Copyright (c) Jupyter Development Team.
+FROM jupyter/all-spark-notebook:dbb9c7b50531
+
+MAINTAINER Jupyter Project <jupyter@googlegroups.com>
+
+# Install Kernel Gateway
+RUN pip install jupyter_kernel_gateway==0.1.0
```

Comment:
Because you're inheriting from jupyter/all-spark-notebook, you should only need to include the RUN, the MAINTAINER, and this RUN. But you need to switch to `USER jovyan` before the RUN so that it installs with proper permissions set for the non-privileged user in the container image and set it back to `USER root` after.

Everything else below this line (and all the associated files) shouldn't be necessary. You can remove the README too since this is meant as a demo at the moment more than anything.